plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/preferences/SSLCetrificatesPreference.java

```diff
@@ -0,0 +1,173 @@
+package org.jboss.tools.openshift.internal.ui.preferences;
+
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import org.eclipse.jface.preference.IPreferenceStore;
+import org.jboss.tools.openshift.common.core.utils.X509CertificateParser;
+import org.jboss.tools.openshift.internal.ui.OpenShiftUIActivator;
+
+public class SSLCetrificatesPreference {
+	final static String ALLOWED_SERTIFICATES = "allowed_sertificates";
+	final static String SEPARATOR = ";";
+
+	static SSLCetrificatesPreference instance = new SSLCetrificatesPreference();
+
+	public static SSLCetrificatesPreference getInstance() {
+		return instance;
+	}
+
+	private List<Item> savedItems = null;
+
+	private SSLCetrificatesPreference() {}
+
+	protected IPreferenceStore getPreferenceStore() {
+		return OpenShiftUIActivator.getDefault().getCorePreferenceStore();
+	}
+
+	/**
+	 * Called by SSLCertificateCallback to retrieve the choice made in the dialog.
+	 * 
+	 * @param certificate
+	 * @return
+	 */
+	public Boolean getAllowedByCertificate(X509Certificate certificate) {
+		X509CertificateParser p = new X509CertificateParser(certificate);
+		Item i = new Item(false, p.getIssuer(), p.getValidity(), p.getFingerprint());
+		Item item = findItem(i);
+		return item != null ? item.checked : null;
+	}
+
+	/**
+	 * Called by SSLCertificateCallback to remember the choice made in the dialog.
+	 *  
+	 * @param certificate
+	 * @return
+	 */
+	public void setAllowedByCertificate(X509Certificate certificate, boolean result) {
+		X509CertificateParser p = new X509CertificateParser(certificate);
+		Item i = new Item(result, p.getIssuer(), p.getValidity(), p.getFingerprint());
+		Item item = findItem(i);
+		if(item == null) {
+			savedItems.add(i);
+		} else {
+			item.checked = result;
+		}
+		saveItems();
+	}
+
+	/**
+	 * Called by preference page.
+	 * Each item is stored as  
+	 * 	checked SEPARATOR issuedBy SEPARATOR validity SEPARATOR fingerprint SEPARATOR
+	 * 
+	 * @return
+	 */
+	List<Item> getSavedItems() {
+		if(savedItems == null) {
+			List<Item> savedItems = new ArrayList<>();
+			String value = getPreferenceStore().getString(ALLOWED_SERTIFICATES);
+			if(value != null && value.length() > 0) {
+				String[] tokens = value.split(SEPARATOR);
+				for (int i = 0; i + 4 <= tokens.length;) {
+					boolean checked = "true".equals(tokens[i++]);
+					String issuedBy = tokens[i++];
+					String validity = tokens[i++];
+					String fingerprint = tokens[i++];
+					savedItems.add(new Item(checked, issuedBy, validity, fingerprint));
+				}
+			}
+			this.savedItems = savedItems;
+		}
+		return savedItems;
+	}
+
+	private Item findItem(Item item) {
+		return getSavedItems().stream().filter(i -> i.equals(item)).findFirst().orElse(null);
+	}
+
+	/**
+	 * Called by preference page
+	 * @param items
+	 */
+	void saveItems(List<Item> items) {
+		savedItems.clear();
```

Comment:
Agree, I removed everywhere direct access to the field, it is either to be accessed as
List<Item> savedItems = getSavedItems();
or to receive the list as a parameter.